packages/fileeditor-extension/src/index.ts

```diff
@@ -174,6 +189,95 @@ function activate(app: JupyterLab, registry: IDocumentRegistry, restorer: ILayou
     return tracker.currentWidget !== null;
   }
 
+
+  class CodeSnippet {
```

Comment:
Let's move the structures and functions that parse markdown into code blocks into a separate model in the coreutils package and add tests.